### PR TITLE
WT-5673 Verify durable timestamp that are part of cell

### PR DESCRIPTION
**Pull request by kommiharibabu**
Verify the added durable timestamps that are part of the both
value and address cells.

### CODE REVIEW

File: src/btree/bt_debug.c [hunk 1]
```diff
@@ -757,7 +761,12 @@ __wt_debug_cursor_hs(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor)
     WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &start.timestamp, &start.txnid,
       &stop.timestamp, &stop.txnid));
 
-    WT_ERR(__debug_time_pairs(ds, "T", start.timestamp, start.txnid, stop.timestamp, stop.txnid));
+    /*
+     * FIXME: Currently, this function is not called anywhere in the code. In future if it is used,
+     * it needs some fixes related to accessing the proper key and value of history store.
+     */
+    // WT_ERR(__debug_time_pairs(ds, "T", start.timestamp, start.txnid, stop.timestamp,
+    // stop.txnid));
```
Comment (agorrod) [inline src/btree/bt_debug.c:769]: Could you explain why this isn't working now?
The comment you've used to disable the code isn't part of our coding standard.

Comment (kommiharibabu) [reply src/btree/bt_debug.c]: The retrieval of the history store key/value are not updated to the latest format. I fixed them now and removed the FIXME and code commenting.

File: src/btree/bt_debug.c [hunk 2]
```diff
@@ -1441,7 +1442,9 @@ __debug_cell(WT_DBG *ds, const WT_PAGE_HEADER *dsk, WT_CELL_UNPACK *unpack)
     case WT_CELL_VALUE_OVFL:
     case WT_CELL_VALUE_OVFL_RM:
     case WT_CELL_VALUE_SHORT:
-        WT_RET(ds->f(ds, ", ts/txn %s,%s",
+        WT_RET(ds->f(ds, ", ts/txn %s,%s,%s,%s",
```
Comment (tetsuo-cpp) [inline src/btree/bt_debug.c:1445]: Just an idea, but should the string contain some information on which timestamp is which? I think start and stop are straightforward but it's less obvious when there are 4 sets of numbers.

Comment (kommiharibabu) [reply src/btree/bt_debug.c]: Yes, I added some more details to differentiate those numbers.

File: src/btree/bt_vrfy.c [hunk 3]
```diff
@@ -1014,28 +1031,36 @@ __verify_page_content(
...
+                                              "oldest start timestamp %s newer than "
+                                              "its newest start durable timestamp %s",
+                  cell_num - 1, __verify_addr_string(session, ref, vs->tmp1),
+                  __wt_timestamp_to_string(unpack.oldest_start_ts, ts_string[0]),
+                  __wt_timestamp_to_string(unpack.newest_start_durable_ts, ts_string[1]));
+            if (unpack.newest_stop_ts != WT_TS_MAX &&
+              unpack.newest_stop_ts > unpack.newest_stop_durable_ts)
+                WT_RET_MSG(session, WT_ERROR, "cell %" PRIu32
+                                              " on page at %s has an "
+                                              "newest stop timestamp %s newer than "
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/btree/bt_vrfy.c]: Fixed.

File: src/btree/bt_vrfy.c [hunk 4]
```diff
@@ -413,6 +413,23 @@ __verify_addr_ts(WT_SESSION_IMPL *session, WT_REF *ref, WT_CELL_UNPACK *unpack,
...
+          "timestamp (%" PRIu64
+          ") newer than its newest start durable "
+          "timestamp (%" PRIu64 ")",
+          __verify_addr_string(session, ref, vs->tmp1), unpack->oldest_start_ts,
+          unpack->newest_start_durable_ts);
+    if (unpack->newest_stop_ts != WT_TS_MAX &&
+      unpack->newest_stop_ts > unpack->newest_stop_durable_ts)
+        WT_RET_MSG(session, WT_ERROR,
+          "internal page reference at %s has an newest stop "
+          "timestamp (%" PRIu64
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/btree/bt_vrfy.c]: Fixed.

**Comment by kommiharibabu** [comment]
The failed test doesn't have any relation to the changes, so merging it.

**Pull request merged**